bazaarvoice/emodb
# Remove Joda

## Github Issue #

None

## What Are We Doing Here?

This is the first PR of several with the overall goal of reducing as many dependencies as possible in the EmoDB client libraries.  This PR in specific focuses only on the removal of Joda time.  Note that this does not mean that Joda is no longer part of the EmoDB classpath since it comes with DropWizard, but it does mean that Joda is no longer used.  This means at a future date when DropWizard is removed as a dependency for the client libraries there won't be any compilation issues due to Joda's removal.

Although the goal is only to remove Joda from the client libraries because many of the API classes are used throughout EmoDB it was simpler to just remove it entirely.  This means a LOT of files were updated but most of these are very straightforward, such as `Duration.standardSeconds(10)` becoming `Duration.ofSeconds(10)`.  For this reason this PR is split into 2 commits.  The first commit contains files where something non-trivial was changed.  This doesn't mean the change is complex, it just means it's more than just a replacement of Joda classes with java.time equivalents.  The second much-larger commit contains all of the trivial updates caused by the nominally different APIs between Joda and java.time.

Generally the following changes were made:

-  Joda `DateTime` became `java.time.Instant` for internal classes and `java.util.Date` for APIs.  My reasoning is that this is not a complete rewrite of EmoDB's time structures and since the existing pattern in the APIs was to use `Date`s I kept that consistent.
- Joda `Duration` became `java.time.Duration`.
- Joda `Period` doesn't have the same meaning in java.time and was also replaced with the closer equivalence of `java.time.Duration`.
- Joda `DateTimeFormatter` became `java.time.format.DateTimeFormatter`
- Joda `Interval` has no java.time equivalence and was replaced with Guava's `Range`.

## How to Test and Verify

Other than the APIs which exposed `DateTime` classes nothing should change.  The best way to verify is to test that EmoDB works as expected using both the existing API clients and the updated versions in the PR.  EmoDB gatekeeper is a good way to verify this.

## Risk

Since very little logic actually changes for the size of this PR it's pretty low risk.  The biggest risk is that an API call may no longer work correctly due to variations in parsing Joda vs java.time objects.

### Level 

`Medium` Again, despite the size of the PR most changes are trivial and should have to functional change.

### Required Testing

`Regression`

## Code Review Checklist


build and run.

we inevitably come back to this code it will only take hours to figure out, not
days.

a victim of rampaging consistency, and should be using this course of action. 
We don't have coding standards out yet for this project, so please make sure to address any feedback regarding STYLE so the codebase remains consistent.
